sklearn/cluster/tests/test_k_means.py

```diff
@@ -892,4 +902,113 @@ def test_less_centers_than_unique_points():
     # centers have been used
     msg = ("Number of distinct clusters (3) found smaller than "
            "n_clusters (4). Possibly due to duplicate points in X.")
-    assert_warns_message(ConvergenceWarning, msg, k_means, X, n_clusters=4)
+    assert_warns_message(ConvergenceWarning, msg, k_means, X,
+                         sample_weights=None, n_clusters=4)
+
+
+def _sort_cluster_centers_and_labels(centers, labels):
+    sort_index = np.argsort(centers, axis=0)[:, 0]
+    sorted_labels = np.zeros_like(labels)
+    for i, l in enumerate(sort_index):
+        sorted_labels[labels == l] = i
+    return centers[sort_index, :], sorted_labels
+
+
+def test_k_means_weighted_vs_repeated():
```

Comment:
can we please parametrize or loop these tests to avoid repeating code for KMeans and MBKMeans?